spec/solver_spec.rb

```diff
@@ -0,0 +1,28 @@
+require_relative 'spec_helper'
+
+describe Solver do
+  before :each do
+    @solver = Solver.new
+  end
+  describe '#solver' do
+    it 'should be an instance of a class solver' do
+      expect(@solver).to be_instance_of Solver
+    end
+  end
+
+  describe '#factorial' do
+    it 'should receive a method factorial and return factorial of n' do
+      expect(@solver.factorial(5)).to eql(120)
+    end
```

Comment:
- [ ] the factorial method should have tests that handle: 
  - Negative value which raises an exception as an expectation
  - Special case - when input is 0, the result should equal 1